test_provider.py

```diff
@@ -0,0 +1,454 @@
+import json
+from datetime import datetime, timedelta
+
+from astropy.coordinates.baseframe import _representation_deprecation
+from common.api import ProgramProvider
+from common.minimodel import *
+from enum import Enum
+from typing import NoReturn, Tuple
+
+    
+class JsonProvider(ProgramProvider):
+
+    obs_classes = {'partnerCal': ObservationClass.PARTNER_CAL,
+                   'science': ObservationClass.SCIENCE,
+                   'programCal': ObservationClass.PROG_CAL,
+                   'acq': ObservationClass.ACQ,
+                   'acqCal': ObservationClass.ACQ_CAL,
+                   'dayCal': None}
+
+    program_types = {'C': ProgramTypes.C,
+                     'CAL': ProgramTypes.CAL,
+                     'DD': ProgramTypes.DD,
+                     'DS': ProgramTypes.DS,
+                     'ENG': ProgramTypes.ENG,
+                     'FT': ProgramTypes.FT,
+                     'LP': ProgramTypes.LP,
+                     'Q': ProgramTypes.Q,
+                     'SV': ProgramTypes.SV}
```

Comment:
In this case, I don't think we need this `dict` since the correspondence is exact between the `ProgramTypes` and the codes. If we have a `str` value, say `program_type`, we can just do:
```
ProgramTypes[program_type]
```
and then avoid the `dict` completely.